### PR TITLE
add default configs for all served runnables

### PR DESCRIPTION
**Pull request by jakerachleff**
Configures the run name to be the path, and some information about the associated repo to be in the metadata of the traced run

### CODE REVIEW

File: langserve/server.py [hunk 1]
```diff
@@ -554,6 +562,38 @@ def _route_name_with_config(name: str) -> str:
     InvokeResponse = create_invoke_response_model(model_namespace, output_type_)
     BatchResponse = create_batch_response_model(model_namespace, output_type_)
 
+    def _update_config_with_defaults(incomingConfig: RunnableConfig) -> RunnableConfig:
```
Comment (eyurtsev) [inline langserve/server.py:565]: Could we move it into a standalone function?

File: langserve/server.py [hunk 2]
```diff
@@ -554,6 +562,38 @@ def _route_name_with_config(name: str) -> str:
...
+
+        # Currently all defaults are non-overridable
+        overridable_default_config = RunnableConfig()
+        
+        metadata = {}
+        
+        is_hosted = os.environ.get("HOSTED_LANGSERVE_ENABLED", "false").lower() == "true"
+        if is_hosted:
+            hosted_metadata = {
+                "Commit SHA": os.environ.get("HOSTED_LANGSERVE_GIT_COMMIT", ""),
```
Comment (eyurtsev) [inline langserve/server.py:576]: Should the keys follow a particular format for rendering purposes / name collision purposes?

File: langserve/server.py [hunk 3]
```diff
@@ -684,6 +725,7 @@ async def batch(
             {k: v for k, v in config_.items() if k in config_keys}
             for config_ in get_config_list(configs, len(inputs_))
         ]
+        print(configs_)
```
Comment (eyurtsev) [inline langserve/server.py:728]:
```suggestion
```

**Review by eyurtsev — COMMENTED** [review]
looks

**Comment by eyurtsev** [comment]
This looks pretty reasonable to me

File: langserve/server.py (hunk 2 above)
Comment (jakerachleff) [reply langserve/server.py:576]: I'm not sure, I started a thread internally to discuss the rendering questions. My worry on naming collision proof names is that they will be hard to read and thus hard to filter on

File: langserve/server.py [hunk 4]
```diff
@@ -122,6 +123,45 @@ def _unpack_request_config(
     )
 
 
+def _update_config_with_defaults(
+    path: str, incomingConfig: RunnableConfig, endpoint: Optional[str] = None
```
Comment (eyurtsev) [inline langserve/server.py:127]:
```suggestion
    path: str, incomingConfig: RunnableConfig, *, endpoint: Optional[str] = None
```

File: langserve/server.py [hunk 5]
```diff
@@ -122,6 +123,45 @@ def _unpack_request_config(
...
+
+    metadata = {}
+
+    if endpoint:
+        metadata["LangServe Endpoint"] = endpoint
+
+    is_hosted = os.environ.get("HOSTED_LANGSERVE_ENABLED", "false").lower() == "true"
+    if is_hosted:
+        hosted_metadata = {
+            "Commit SHA": os.environ.get("HOSTED_LANGSERVE_GIT_COMMIT", ""),
```
Comment (eyurtsev) [inline langserve/server.py:142]: The naming convention for these is different from langserve version etc. Should we use the same style? (I don't care which style is used)

File: langserve/server.py [hunk 6]
```diff
@@ -555,7 +601,7 @@ def _route_name_with_config(name: str) -> str:
     BatchResponse = create_batch_response_model(model_namespace, output_type_)
 
     async def _get_config_and_input(
-        request: Request, config_hash: str
+        request: Request, config_hash: str, endpoint: Optional[str] = None
```
Comment (eyurtsev) [inline langserve/server.py:604]:
```suggestion
        request: Request, config_hash: str, *, endpoint: Optional[str] = None
```

File: langserve/server.py [hunk 7]
```diff
@@ -735,7 +787,7 @@ async def stream(
         err_event = {}
         validation_exception: Optional[BaseException] = None
         try:
-            config, input_ = await _get_config_and_input(request, config_hash)
+            config, input_ = await _get_config_and_input(request, config_hash, "stream")
```
Comment (eyurtsev) [inline langserve/server.py:790]:
```suggestion
            config, input_ = await _get_config_and_input(request, config_hash, endpoint="stream")
```

File: langserve/server.py [hunk 8]
```diff
@@ -474,6 +515,11 @@ def add_routes(
             f"If specifying path please start it with a `/`"
         )
 
+    if "run_name" in config_keys:
+        raise ValueError(
+            "Cannot configure run_name. Please remove it from config_keys."
```
Comment (eyurtsev) [inline langserve/server.py:520]: Do we care about key collisions arising from identical path on 2 different API routers? (I think it's fine actually)

**Review by eyurtsev — APPROVED** [review]

File: langserve/server.py (hunk 5 above)
Comment (jakerachleff) [reply langserve/server.py:142]: what is the current convention?

Comment (jakerachleff) [reply langserve/server.py:142]: just `__langserve`?

File: langserve/server.py (hunk 8 above)
Comment (jakerachleff) [reply langserve/server.py:520]: oh this is a good point... we probably should?

Comment (jakerachleff) [reply langserve/server.py:513]: We derive it from `request.url.path` and always get the config based on the request?

Comment (jakerachleff) [reply langserve/server.py:513]: might limit future support for complex paths tho

File: langserve/server.py (hunk 5 above)
Comment (eyurtsev) [reply langserve/server.py]: yeah but whatever you prefer -- i just chose arbitrarily

File: langserve/server.py (hunk 8 above)
Comment (eyurtsev) [reply langserve/server.py:513]: I'm OK with whatever you choose here -- just wanted to surface that collisions could potentially appear

File: langserve/server.py [hunk 9]
```diff
@@ -132,6 +133,56 @@ def _unpack_request_config(
     )
 
 
+def _update_config_with_defaults(
+    path: str,
+    incomingConfig: RunnableConfig,
```
Comment (eyurtsev) [inline langserve/server.py:138]:
```suggestion
    incoming_config: RunnableConfig,
```

Comment (eyurtsev) [reply langserve/server.py:138]: (python naming convention)

File: langserve/server.py [hunk 10]
```diff
@@ -794,7 +840,9 @@ async def stream(
         err_event = {}
         validation_exception: Optional[BaseException] = None
         try:
-            config, input_ = await _get_config_and_input(request, config_hash)
+            config, input_ = await _get_config_and_input(
+                request, config_hash, endpoint="stream"
```
Comment (eyurtsev) [inline langserve/server.py:844]: are we missing a similar config override for stream_log endpoint?

**Pull request merged**